dataland-frontend/tests/e2e/specs/user-authentication/Register.ts

```diff
@@ -73,4 +74,45 @@ describe("As a user I want to be able to register for an account and be able to
     });
     logout();
   });
+
+  it("Checks that TOTP-Based 2FA can be enabled for the newly registered account", () => {
+    cy.task("getEmail").then((returnEmail) => {
+      cy.task("getPassword").then((returnPassword) => {
+        const username = returnEmail as string;
+        const password = returnPassword as string;
+        login(username, password);
+        cy.visitAndCheckAppMount("/companies")
+          .get("div[id='profile-picture-dropdown-toggle']")
+          .click()
+          .get("a[id='profile-pocture-dropdown-settings-button']")
+          .click()
+          .get("div[id='landing-signingin'] > a")
+          .should("be.visible", { timeout: 20000 })
+          .click()
+          .get("button:contains('Set up authenticator application')")
+          .should("be.visible", { timeout: 20000 })
+          .click()
+          .get("a:contains('Unable to scan')")
+          .should("be.visible", { timeout: 10000 })
+          .click()
+          .get("span[id='kc-totp-secret-key']")
+          .should("be.visible", { timeout: 10000 })
+          .invoke("text")
+          .then((text) => {
+            const totpKey = text.replace(/\s/g, "");
+            cy.get("input[id='totp']")
+              .type(authenticator.generate(totpKey))
+              .get("input[id='saveTOTPBtn']")
+              .click()
+              .get("button[id='signOutButton']")
+              .should("be.visible", { timeout: 20000 });
+
+            logout();
+            login(username, password, () => {
+              return authenticator.generate(totpKey);
+            });
```

Comment:
Something does not work here.
I've added "verifySearchResultsPage()" into "login()" before the url.eq is executed to assure, that the search results are actually displayed at the moment when it is checked that the url is "**/companies". 
Then I ran the test locally, and I observed that it fails because the search results actuall never appear!
Very stragen, but somehow the login does not work and the data cannot be retrieved from the backend, right?
We need to fix this.

Reply:
I tried stabelizing the test by splitting it into 2 different test (one for the setup and one for the login). Meaning that if the login part fails for some bizarre totp-code-timeout, it can be retried. It seems to be relatively stable now.

Reply:
I ran the test two times in Electron.
On both runs, it failed on the first attempt, and succeeded in the second attempt.
It failed on the first attempts, because the otp code, which is typed into the input field, was incorrect during the login process.
I think we should not merge it to main as long as it behaves like this.

Reply:
On my third Electron run, it failed again on the first attempt.
Then I switched to Chrome.
In Chrome it passed on thre runs and always on the frist attempt.

So it seems that it could be a timeout thing again?

Reply:
We could not find the reason behind this.
Anyways, the tests at least always passes on a second attempt. Therefore we decided that it's acceptable to use it like this.